### PR TITLE
Preserve bibtex import order

### PR DESCRIPTION
**Pull request by tobiasdiez**
@obraliar https://github.com/JabRef/jabref/pull/504#discussion_r56398599 noted that the order of the bibentries in `parserResult.getDatabase().getEntries` is not the same as in the bib file.
As it turned out, fixing the order required to store the bibentries not as `Map<id, BibEntry>` but simply as a list. So I removed almost everything which is connected to ID's. Now they are only used for writing bibentries (to determine the original order) and for the sql export. 

- [x] Change in CHANGELOG.md described? Not relevant
- [x] Changes in pull request outlined? (What, why, ...)
- [x] Tests created for changes?
- [x] Tests green?

### CODE REVIEW

File: src/main/java/net/sf/jabref/exporter/ExportFormats.java [hunk 1]
```diff
@@ -223,7 +219,7 @@ public void actionPerformed(ActionEvent e) {
                     Globals.prefs.put(JabRefPreferences.EXPORT_WORKING_DIRECTORY, file.getParent());
 
                     final File finFile = file;
-                    final Set<String> finEntryIDs = entryIds;
+                    final List<BibEntry> finEntries = entries;
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/exporter/ExportFormats.java]: Why is a new variable introduced?

File: src/main/java/net/sf/jabref/logic/labelpattern/LabelPatternUtil.java [hunk 2]
```diff
@@ -499,11 +499,11 @@ public static void makeLabel(MetaData metaData, BibDatabase dBase, BibEntry entr
         if (!alwaysAddLetter && (occurrences == 0)) {
             // No dupes found, so we can just go ahead.
             if (!key.equals(oldKey)) {
-                if (database.getEntryById(entry.getId()) == null) {
+                if (! database.containsEntryWithId(entry.getId())) {
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/logic/labelpattern/LabelPatternUtil.java]: No space...

**Comment by oscargus** [comment]
:+1: (Just one question and one space...)

I guess an intermediate step would be to have a List of ids instead of a Set as that list should probably have a smaller memory footprint compared to bibentry, but as long as it is not a problem (and I guess it isn't, even with thousands of entries), this seems to be the way to go!

File: src/main/java/net/sf/jabref/exporter/ExportFormats.java (hunk 1 above)
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/exporter/ExportFormats.java]: Further down it is used in an anonymous method, which is only possible if it final.

**Comment by simonharrer** [comment]
:+1:

**Pull request merged**